### DISCUSSION TITLE
Prevent dragging of pinned tab

### DISCUSSION THREAD

**Discussion by enko** [piroor/treestyletab#3461] [Q&A] [answered]
Hello 👋,

I accidentally move pinned tabs all the time, but I don't want to. Can I prevent dragging pinned tabs to somewhere else?

Thanks,

Tim

**Comment by irvinm** [comment]
@piroor would know better than me, but I could only think of two things, but neither I think would work.

1. Disable "pointer-events" for pinned tabs.  While this would prevent you from moving the tabs, it would also prevent you from interacting with them with either left or right clicks.  (If you are ok with only keyboard interactions, this might work)
2. There is a property "draggable" set on the tab-item-substance ... if this is toggled to "false" the tab can't be dragged.  (You can try it manually)  The problem with this is that I don't think you can change this value via CSS.  I think @piroor would have to add an option to allow users to enable\disable this feature and handle it in JS.

![image](https://github.com/piroor/treestyletab/assets/979729/aac6bdd5-ca06-4c7d-9b3b-29f4e3814492)

**Reply by enko** [reply]
Setting `draggable` to `false`, works like I want it to do.

**Reply by piroor** [reply]
Before we think about how to do that, we need to know why the demand is posted here. For example, if it is aimed to prevent accidental detaching from the window, it is better solution fixing the too sensitive behavior instead of disabling draggability of pinned tabs. Could you add such an information? @enko

**Reply by enko** [reply]
@piroor it is actually about the sensitivity. Every so often I click on a tab, and it just detaches. For me, this is especially bad because I use a tiling window manager with only full-screen layout, and it is challenging to drag the window back to the original window. Since I would rather not move pinned tabs somewhere else because I pinned them.

**Reply by piroor** [reply, marked as answer]
There is an expert option: TST options => check "Unlock Expert Options" => "Drag and Drop" => choose "Do nothing " for "Drag" action, then TST won't detach dragged tabs from the window.
Due to some unfixed edge-case issue around IPC on Firefox itself, sometimes drag session is unexpectedly left living after you release the mouse button and it is terminated on the next starting of a new drag session, thus such a finished zombie drag session firs "dragend" event and causes such an unexpected detaching of a clicked tab. Sadly we addon author side can do nothing about such a bug and just disabling the feature may be only one solution when you see the problem too frequently.

**Reply by enko** [reply]
That's lovely, especially because shift + drag is still an option if I need to drag something.

Thanks for your time and also explaining the actual issue to me, which sounds a lot more what happened to me.
